packages/xstate-vue/src/useService.ts

```diff
@@ -0,0 +1,31 @@
+import { ref, Ref, onBeforeUnmount } from '@vue/composition-api';
+import { EventObject, State, Interpreter } from 'xstate';
+
+export function useService<TContext, TEvent extends EventObject>(
+  service: Interpreter<TContext, any, TEvent>
+): {
+  current: Ref<State<TContext, TEvent>>;
+  send: Interpreter<TContext, any, TEvent>['send'];
+  service: Ref<Interpreter<TContext, any, TEvent>>;
+} {
+  const current = ref<State<TContext, TEvent>>(service.state);
+  const serviceRef = ref<Interpreter<TContext, any, TEvent>>(service);
+
+  const { unsubscribe } = serviceRef.value.subscribe(state => {
```

Comment:
In other hooks you subscribe in `onBeforeMount` and here you subscribe directly in the setup function. Could you elaborate more on that? Is this safe?

Reply:
You are right on that, as I'm not that familiar with React hooks I just translated `useEffect` to those Lifecycle hooks in my mind. I think generally using the `onBeforeMount` hooks are not needed, as the state machines don't interact with the DOM directly. (where `onMounted` would be needed e.g.)

Reply:
Right - I'm just wondering if it's possible for a setup function to be called but a corresponding component never actually gets mounted like rendered in memory but discarded shortly after that. Do you know if it's possible in Vue?

Reply:
There are [Renderless Components](https://adamwathan.me/renderless-components-in-vuejs/) in Vue, although they are also bound to the lifecycle of the component that is using them.